pkg/cluster/k8sres.go

```diff
@@ -1156,6 +1176,8 @@ func (c *Cluster) generateStatefulSet(spec *acidv1.PostgresSpec) (*appsv1.Statef
 		initContainers,
 		sidecarContainers,
 		&tolerationSpec,
+		effectiveRunAsUser,
```

Comment:
My only concern is here, this seems to set runAsUser and runAsGroup to the new 101/103 combination by default.

I will have to understand what this was/is before.